src/bp/core/server.ts

```diff
@@ -235,4 +239,51 @@ export default class HTTPServer {
       }
     }
   }
+
+  extractExternalToken = async (req, res, next) => {
+    if (req.headers.externalauth) {
+      try {
+        req.credentials = await this.decodeExternalToken(req.headers.externalauth)
+      } catch (error) {
+        return next(new InvalidExternalToken(error.message))
+      }
+    }
+
+    next()
+  }
+
+  async decodeExternalToken(externalToken): Promise<any | undefined> {
+    const { enabled, publicKey, audience, algorithm } = await this._getExternalAuthConfig()
+
+    if (!enabled) {
+      return undefined
+    }
+
+    const [scheme, token] = externalToken.split(' ')
```

Comment:
we do this in couple of places, can we extract this in a utils.security package or something like that ?

Reply:
We do this twice, for about 5 lines in two different parts. It could be merged, but is it really necessary?